scripts/tools/RegisterPublishedDeliveryWebhook.php

```diff
@@ -0,0 +1,131 @@
+use oat\tao\model\webhooks\WebhookRegistryManager;
+use oat\tao\model\webhooks\WebhookRegistryManagerInterface;
+use oat\taoPublishing\model\publishing\event\RemoteDeliveryCreatedEvent;
+
+class RegisterPublishedDeliveryWebhook extends ScriptAction
+{
+    use ServiceManagerAwareTrait;
+
+    private const DEFAULT_HTTP_METHOD = 'GET';
+
+    const ACCEPTED_HTTP_METHOD = [
+        'POST',
+        self::DEFAULT_HTTP_METHOD
+    ];
+    /** @var Report */
+    private $report;
+
+    protected function provideOptions()
+    {
+        return [
+            'simpleRosterUrl' => [
+                'prefix' => 'u',
+                'longPrefix' => 'url',
+                'flag' => false,
+                'description' => 'Will register endpoint to inform about new deliveries published',
+                'required' => true
+            ],
+            'httpMethod' => [
+                'prefix' => 'm',
+                'description' => 'Determine what type of http method use',
+                'flag' => false,
+                'required' => false
+            ]
+        ];
+    }
+
+    protected function provideDescription()
+    {
+        return 'Script will register webhook with a defined url';
+    }
+
+    protected function run(): Report
+    {
+        $this->report = Report::createInfo('Registering webhook');
+
+        /** @var ConfigurableService $eventManager */
+        $eventManager = $this->getServiceLocator()->get(EventManager::class);
+        /** @var ConfigurableService $webhookEventsService */
+        $webhookEventsService = $this->getServiceLocator()->get(WebhookEventsService::class);
+
+        try {
+            $webhookEventsService->registerEvent(RemoteDeliveryCreatedEvent::class);
+            $this->getServiceManager()->register(WebhookEventsService::SERVICE_ID, $webhookEventsService);
+            $this->getServiceManager()->register(EventManager::SERVICE_ID, $eventManager);
+        } catch (Exception $exception) {
+            $this->report->add(
+                Report::createError('Registering failed')
+            );
+        }
+
+        $this->getWebhookRegistryManager()->addWebhookConfig(
+            $this->createWebHook(),
+            RemoteDeliveryCreatedEvent::class
+        );
+
+        return $this->report;
+    }
+
+    private function getWebhookRegistryManager(): WebhookRegistryManagerInterface
+    {
+        return $this->getServiceLocator()->get(WebhookRegistryManager::class);
+    }
+
+    private function getHttpMethod(): string
+    {
+        $method = $this->getOption('httpMethod') ?? self::DEFAULT_HTTP_METHOD;
+        if (!in_array($method, self::ACCEPTED_HTTP_METHOD)) {
+            $this->report->add(
+                Report::createWarning(sprintf("Used illegal method %s. Falling back to default method", $method))
+            );
+            return self::DEFAULT_HTTP_METHOD;
+        }
+
+        return $method;
+    }
+
+    private function createWebHook()
```

Comment:
Return types
also here:
https://github.com/oat-sa/extension-tao-publishing/pull/97/files#diff-6718de02cddabc7d08a6eb05e47fbc30b3a8b9042d438eaab02f8bb62cdaed25R70
https://github.com/oat-sa/extension-tao-publishing/pull/97/files#diff-6718de02cddabc7d08a6eb05e47fbc30b3a8b9042d438eaab02f8bb62cdaed25R51

Reply:
These methods are extended from [this abstract class](https://github.com/oat-sa/generis/blob/master/common/oatbox/extension/script/ScriptAction.php#L32) and I don't think it's a good idea to declare return types for them (because in `parent` they are not declared in current implementation).

Also here: [run()](https://github.com/oat-sa/extension-tao-publishing/pull/97/files#diff-6718de02cddabc7d08a6eb05e47fbc30b3a8b9042d438eaab02f8bb62cdaed25R75)